waspc/data/Generator/templates/server/src/core/auth.js

```diff
@@ -69,5 +70,52 @@ export const verifyPassword = async (hashedPassword, password) => {
   }
 }
 
-export default auth
+// Generates an unused username that looks similar to "quick-purple-sheep-91231". 
+// It generates several options and ensures it picks one that is not currently in use.
+export function generateAvailableDictionaryUsername() {
+  const adjectives = ['fuzzy', 'tall', 'short', 'nice', 'happy', 'quick', 'slow', 'good', 'new', 'old', 'first', 'last', 'old', 'young']
+  const colors = ['red', 'green', 'blue', 'white', 'black', 'brown', 'purple', 'orange', 'yellow']
+  const nouns = ['cat', 'dog', 'lion', 'rabbit', 'duck', 'pig', 'bee', 'goat', 'crab', 'fish', 'chicken', 'horse', 'llama', 'camel', 'sheep']
+
+  const potentialUsernames = []
+  for (let i = 0; i < 10; i++) {
+    const potentialUsername = `${adjectives[randomInt(adjectives.length)]}-${colors[randomInt(colors.length)]}-${nouns[randomInt(nouns.length)]}-${randomInt(100_000)}`
+    potentialUsernames.push(potentialUsername)
+  }
+
+  return findAvailableUsername(potentialUsernames)
+}
+
+// Generates an unused username based on an array of username segments and a separator. 
+// It generates several options and ensures it picks one that is not currently in use.
+export function generateAvailableUsername(usernameSegments, config) {
+  const separator = config?.separator || '-'
+  const baseUsername = usernameSegments.join(separator)
+
+  const potentialUsernames = []
+  for (let i = 0; i < 10; i++) {
+    const potentialUsername = `${baseUsername}${separator}${randomInt(100_000)}`
+    potentialUsernames.push(potentialUsername)
+  }
+
+  return findAvailableUsername(potentialUsernames)
+}
+
+// Checks the database for an unused username from an array provided and returns first.
+async function findAvailableUsername(potentialUsernames) {
+  const users = await prisma.{= userEntityLower =}.findMany({
+    where: {
+      username: { in: potentialUsernames },
+    }
+  })
+  const takenUsernames = users.map(user => user.username)
```

Comment:
You mean just `select`ing the username col, instead of returning the whole object? Yeah, we could. At max we will have just 10 results though, so I am not sure it would be a huge gain.